File: example1.m
% régression linéaire 

% création du modèle :
model = @(param, x) param(1) * x + param(2);


% génération des données :
x = linspace(-5, 15, 120);
y = 4.5 * x + 12 + randn(size(x));



% fit :
p = metropolis(model, x, y, [5, 10], [0.5, 1], 2000);


% moyennes et écarts types :
a = mean(p(500:end, 1));
da = std(p(500:end, 1));
b = mean(p(500:end, 2));
db = std(p(500:end, 2));

disp(["pente = ", num2str(a), " +- ", num2str(da)]);
disp(["ordonnée à l'origine = ",num2str(b) " +- ", num2str(db)]);

File: example1.py
import metropolis
import numpy as np

# régression linéaire 


# création du modèle :
def model(p, x):
	# p[0] is used internaly
	return p[1] * x + p[2]

# génération aléatoire des données :
x = np.linspace(-5, 15, 120)
y = 4.5 * x + 12 + np.random.randn(len(x))

# fit :
p = metropolis.metropolis(model, x, y, [5, 10], [0.5, 1])


# moyennes et écarts types :
a  = np.mean(p[:,1], axis=0)
da = np.std( p[:,1], axis=0)
b  = np.mean(p[:,2], axis=0)
db = np.std( p[:,2], axis=0)

print "pente = {} +- {}".format(a, da)
print "ordonnee a l'origine = {} +- {}".format(b, db)
